### DISCUSSION TITLE
Typing Status

### DISCUSSION THREAD

**Discussion by ajbott** [PullDating/Org#198] [Ideas & Features]
## Overview

The Typing Status feature aims to enhance the communication experience within the app by providing real-time indicators when a conversation partner is typing. This functionality is similar to platforms like Discord, where users receive notifications when someone is actively composing a message.

### Improved Communication Flow:
- Real-time typing indicators create a more dynamic and engaging conversation environment.
- Users can anticipate responses, fostering a sense of responsiveness and connection.

### Increased Patience:
- Knowing that the partner is actively typing encourages users to wait for a thoughtful response, reducing impatience.

## Implementation

- When a user starts typing, a visual indicator (e.g., ellipsis or "typing" status) appears for their conversation partner.
- The indicator disappears when the user stops typing.
- This feature enhances the haptic communication experience and contributes to more meaningful interactions.

## User Experience

- Users can enjoy a more immersive and responsive chat experience with the introduction of the Typing Status feature.
- The visual cues provide valuable insights into the timing and flow of the conversation.

## Conclusion

The Typing Status feature is designed to enrich the messaging experience, fostering improved communication dynamics and encouraging users to engage in more thoughtful and patient conversations.

**Comment by flo-kei** [comment]
good feature 👍🏻 

however i would **NOT** add a "X is typing..." experience like on snapchat, where you get a notification each time someone starts typing a reply to your message (i find that annoying as hell, and it draws way too much attention to the app)

**Comment by Oshyan** [comment]
I'm going to push back on the idea of implementing this feature a bit. We are *not* trying to create a general replacement for a chat app, or even a 1:1 match with the *experience* of such an app. A dating app should, IMHO, encourage prompt interaction with current matches, but not be expected to be a priority or put pressure on each person unnecessarily. People who **know each other well** already sometimes indicate feeling pressured by the "typing" indicator in normal messaging apps! The dynamic between people who *don't* know each other is likely to be much more fraught and unpredictable. 

It would also be a technical and potential resource-use challenge since it requires live and more frequent server-app interaction. Since we are trying to keep costs low an unnecessarily "expensive" feature like this doesn't seem like it is worthwhile enough overall to justify its downsides. So I don't see a significant reason to implement this.

Perhaps we can step back and consider what problem or frustration we might be trying to solve in implementing this feature, and then think if there are other ways to partially or fully solve that with less problematic methods. For example if we're considered about responsiveness, incentivizing that more broadly (as recently touched on in other discussions) might be a better, albeit indirect, way to support and encourage that without most of the downsides of an actual, realtime typing indicator.

**Comment by ajbott** [comment]
I personally wouldn't use this feature. Having to switch to another messaging app to compose a message to copy and paste into the app seems like a hassle, and it usually raises questions on the other person's end.

I believe there are more pressing features to prioritize before considering this one. However, if it could be implemented seamlessly without any drawbacks, such as avoiding direct connections between users, it might be worth considering in the future. Also, it would still be off by default. If people want to turn it on they can. Has to be low maintenance (if it breaks, it breaks).

**Reply by Oshyan** [reply]
I think in general we need to be a lot more decisive with things like this. If it's something that people have to turn on (and presumably *both* parties have to have it on for it to work), and there are the downsides I mentioned above (which weren't entirely addressed, did you agree with them?), I really don't see a future in which it should be implemented. I mean a clear case wasn't really made *in favor* of it, IMO. Critically, dating is not like Discord and not like most other contexts of communication either. The emotional context and weight are different and such features need to be considered *in this context* to understand if they would be valuable. It's not enough to know that they are considered valuable in other contexts, IMO.
